pkg/operation/botanist/secrets.go

```diff
@@ -459,7 +486,11 @@ func (b *Botanist) SyncShootCredentialsToGarden(ctx context.Context) error {
 				secretObj.Annotations = s.annotations
 				secretObj.Labels = s.labels
 				secretObj.Type = corev1.SecretTypeOpaque
-				secretObj.Data = b.LoadSecret(s.secretName).Data
+				if s.secretName == v1beta1constants.SecretNameOldSSHKeyPair {
+					secretObj.Data = oldSecret.Data
+				} else {
+					secretObj.Data = b.LoadSecret(s.secretName).Data
+				}
```

Comment:
Why don't you simply run `b.StoreSecret` above when the old secret was found? Then you can keep this code untouched.

Reply:
Addressed in [commit](https://github.com/gardener/gardener/pull/5388/commits/8e9970ad6576f23a3b8b79382aa9c8542ddd7ad4).